### DISCUSSION TITLE
Gmx dipoles doesnt work

### DISCUSSION THREAD

**Discussion by jshinsato1** [Becksteinlab/GromacsWrapper#291] [Q&A]
when using gromacs.dipoles it produces an error, I assume that this is because when run normally through the terminal gmx dipoles prompts the user to select a group to analyze, similar to how gmx energy or gmx pdb2gmx would do. Is there a workaround for this or will commands like that simply not work.
![gmxdipoles](https://github.com/user-attachments/assets/e98a2f7a-52fa-4ad0-bd36-acdf52e43c1a)

**Comment by orbeckst** [comment]
If a command requires input then in gromacswrapper use the `input=[...]` kwarg as described under [Simple Usage](https://gromacswrapper.readthedocs.io/en/latest/overview.html#simple-usage):
```python
gromacs.dipoles(corr="mol", P=1, temp=298, 
                f="md0.trr", s="md0.tpr", 
                o="Mtot0.xvg", d="dipdist0.xvg", c="dipcorr0.xvg",
                input=["Water"])
```
I didn't try it but something like the above should work. It's the equivalent of piping into the command from the command line
```bash
printf "Water\n" | gmx dipoles -corr mol -P 1 -s md0.tpr ...
```

**Reply by orbeckst** [reply]
To see the equivalent shell command to a GromacsWrapper command, [start logging](https://gromacswrapper.readthedocs.io/en/latest/overview.html#logging) with
```python
gromacs.start_logging()
```
